### PR TITLE
Update probs in Lightning.GPU

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

[SC-75562] & [SC-73560]

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by josephleekl** [comment]
This looks good to me, thanks @multiphaseCFD!

**Review by josephleekl — APPROVED** [review]
This looks good to me, thanks @multiphaseCFD!

File: pennylane_lightning/core/_measurements_base.py [hunk 1]
```diff
@@ -131,15 +131,6 @@ def expval(self, measurementprocess: MeasurementProcess):
...
 
-    def _probs_retval_conversion(self, probs_results: Any) -> np.ndarray:
-        """Convert the data structure from the C++ backend to a common structure through lightning devices.
-        Args:
-            probs_result (Any): Result provided by C++ backend.
-        Returns:
-            np.ndarray with probabilities of the supplied observable or wires.
-        """
-        return probs_results
-
```
Comment (LuisAlfredoNu) [inline pennylane_lightning/core/_measurements_base.py:142]: It is good to remove this extra function from the base class

File: tests/lightning_qubit/test_measurements_class.py [hunk 2]
```diff
@@ -669,7 +669,7 @@ def test_double_return_value(self, shots, measurement, obs0_, obs1_, lightning_s
             assert np.allclose(r, e, atol=dtol, rtol=dtol)
 
     @pytest.mark.skipif(
-        device_name in ("lightning.gpu", "lightning.tensor"),
+        device_name in ("lightning.tensor"),
```
Comment (LuisAlfredoNu) [inline tests/lightning_qubit/test_measurements_class.py:672]:
```suggestion
        device_name == "lightning.tensor",
```

File: tests/test_measurements.py [hunk 3]
```diff
@@ -151,8 +151,8 @@ def circuit():
             _ = circuit()
 
     @pytest.mark.skipif(
-        device_name in ("lightning.gpu", "lightning.tensor"),
-        reason="lightning.gpu/lightning.tensor does not support out of order prob.",
+        device_name in ("lightning.tensor"),
```
Comment (LuisAlfredoNu) [inline tests/test_measurements.py:154]:
```suggestion
        device_name == "lightning.tensor",
```

**Review by LuisAlfredoNu — APPROVED** [review]
Thank you @multiphaseCFD 
Nice feature addition to LGPU!
Just a few suggestions but feel free to ignore them if you need it.

**Pull request merged**